front/app/modules/commercial/user_custom_fields/admin/components/GenderChart.tsx

```diff
@@ -163,18 +194,30 @@ class GenderChart extends PureComponent<Props & WrappedComponentProps, State> {
             <PieChartStyleFixesDiv>
               <PieChart
                 data={serie}
+                width={164}
                 mapping={{
                   angle: 'value',
                   name: 'name',
-                  fill: ({ row }) => labelColors[row.code] ?? colorMain,
+                  opacity: ({ rowIndex }) => {
+                    if (this.state.hoverIndex === undefined) return 1;
+                    return this.state.hoverIndex === rowIndex ? 1 : 0.3;
+                  },
                 }}
                 pie={{
                   startAngle: 0,
                   endAngle: 360,
                   outerRadius: 60,
                 }}
                 innerRef={this.currentChart}
-                annotations
+                tooltip={renderTooltip()}
+                legend={{
+                  items: legend as LegendItem[],
```

Comment:
You should avoid using `as`, it basically overrides TypeScript completely. In this case, `legend` can be `LegendItem[] | null`. `items` does not accept `null`, which TS will tell you if you don't use `as`.